### PR TITLE
HTTP/3: Log frames and remove frame max size

### LINKED ISSUE #29035: [HTTP/3] Remove maxFrameSize in HTTP/3
There isn't a max frame size in HTTP/3, the only max size is for headers, the HEADER_LIST size. We should remove it from https://github.com/dotnet/aspnetcore/blob/master/src/Servers/Kestrel/Core/src/Internal/Http3/Http3FrameWriter.cs#L36

### LINKED ISSUE #29080: HTTP/3 logging
Improve HTTP/3 logging to be on par with 1.1 and 2.

Ensure all log messages are defined, log incoming and outgoing frames, etc.

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspnetcore/issues/29080
Fixes https://github.com/dotnet/aspnetcore/issues/29035

### CODE REVIEW

File: src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs [hunk 1]
```diff
@@ -138,6 +138,14 @@ internal class KestrelTrace : IKestrelTrace
             LoggerMessage.Define<string, Http3ErrorCode>(LogLevel.Debug, new EventId(45, "Http3StreamAbort"),
                 @"Trace id ""{TraceIdentifier}"": HTTP/3 stream error ""{error}"". An abort is being sent to the stream.");
 
+        private static readonly Action<ILogger, string, Http3FrameType, long, long, Exception?> _http3FrameReceived =
+            LoggerMessage.Define<string, Http3FrameType, long, long>(LogLevel.Trace, new EventId(46, "Http3FrameReceived"),
+                @"Connection id ""{ConnectionId}"" received {type} frame for stream ID {id} with length {length}.");
+
+        private static readonly Action<ILogger, string, Http3FrameType, long, long, Exception?> _http3FrameSending =
+            LoggerMessage.Define<string, Http3FrameType, long, long>(LogLevel.Trace, new EventId(47, "Http3FrameReceived"),
```
Comment (halter73) [inline src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs:146]:
```suggestion
            LoggerMessage.Define<string, Http3FrameType, long, long>(LogLevel.Trace, new EventId(47, "Http3FrameSent"),
```

We should do the same for Http2FrameSent.

**Review by halter73 — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs (hunk 1 above)
Comment (jkotalik) [reply src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs:146]: Should we rename _http3FrameSending to _http3FrameSent then?

**Review by jkotalik — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs (hunk 1 above)
Comment (halter73) [reply src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs]: I'm fine with either "sending" or "sent" as long as we're consistent.

Comment (JamesNK) [reply src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs]: IMO sending is more accurate and more consistent 😋

**Pull request merged**